// @vue-macros/setup-block@0.1.1/dist/chunk-DEXDLEV6.mjs
// src/index.ts
import { createUnplugin } from "unplugin";
import { createFilter } from "@rollup/pluginutils";
import { REGEX_SETUP_SFC, REGEX_VUE_SFC } from "@vue-macros/common";

// src/core/index.ts
import { MagicString, getTransformResult } from "@vue-macros/common";
import { parse } from "@vue/compiler-dom";
function transformSetupBlock(code, id, lang) {
  const s = new MagicString(code);
  const node = parse(code, {
    // there are no components at SFC parsing level
    isNativeTag: () => true,
    // preserve all whitespaces
    isPreTag: () => true,
    getTextMode: ({ tag, props }, parent) => {
      if (!parent && tag !== "template" || // <template lang="xxx"> should also be treated as raw text
      tag === "template" && props.some(
        (p) => p.type === 6 && p.name === "lang" && p.value && p.value.content && p.value.content !== "html"
      )) {
        return 2;
      } else {
        return 0;
      }
    }
  });
  for (const child of node.children) {
    if (child.type === 1 && child.tag === "setup") {
      const hasLang = child.props.some((p) => p.name === "lang");
      let codegen = "script setup";
      if (!hasLang && lang) {
        codegen += ` lang="${lang}"`;
      }
      s.overwrite(
        child.loc.start.offset + 1,
        child.loc.start.offset + 6,
        codegen
      );
      s.overwrite(
        child.loc.end.offset - 6,
        child.loc.end.offset - 1,
        "script"
      );
    }
  }
  return getTransformResult(s, id);
}

// src/index.ts
function resolveOption(options) {
  return {
    include: [REGEX_VUE_SFC, REGEX_SETUP_SFC],
    defaultLang: "ts",
    ...options
  };
}
var name = "unplugin-vue-setup-block";
var src_default = createUnplugin(
  (userOptions = {}) => {
    const options = resolveOption(userOptions);
    const filter = createFilter(options.include, options.exclude);
    return {
      name,
      enforce: "pre",
      transformInclude(id) {
        return filter(id);
      },
      transform(code, id) {
        try {
          return transformSetupBlock(code, id, options.defaultLang);
        } catch (err) {
          this.error(`${name} ${err}`);
        }
      }
    };
  }
);

export {
  src_default
};
