// @vue-macros/api@0.4.3/dist/ts.js
"use strict";Object.defineProperty(exports, "__esModule", {value: true});
















var _chunk32PZ64CNjs = require('./chunk-32PZ64CN.js');

















exports.exportsSymbol = _chunk32PZ64CNjs.exportsSymbol; exports.getTSFile = _chunk32PZ64CNjs.getTSFile; exports.isTSDeclaration = _chunk32PZ64CNjs.isTSDeclaration; exports.isTSExports = _chunk32PZ64CNjs.isTSExports; exports.mergeTSProperties = _chunk32PZ64CNjs.mergeTSProperties; exports.resolveTSEntityName = _chunk32PZ64CNjs.resolveTSEntityName; exports.resolveTSExports = _chunk32PZ64CNjs.resolveTSExports; exports.resolveTSFileId = _chunk32PZ64CNjs.resolveTSFileId; exports.resolveTSFileIdNode = _chunk32PZ64CNjs.resolveTSFileIdNode; exports.resolveTSProperties = _chunk32PZ64CNjs.resolveTSProperties; exports.resolveTSReferencedType = _chunk32PZ64CNjs.resolveTSReferencedType; exports.resolveTSScope = _chunk32PZ64CNjs.resolveTSScope; exports.resolveTypeElements = _chunk32PZ64CNjs.resolveTypeElements; exports.setResolveTSFileIdImpl = _chunk32PZ64CNjs.setResolveTSFileIdImpl; exports.tsFileCache = _chunk32PZ64CNjs.tsFileCache; exports.tsFileExportsCache = _chunk32PZ64CNjs.tsFileExportsCache;
exports.default = module.exports;
