// @vue-macros/define-props@0.3.2/dist/chunk-KEQSZVVA.js
"use strict";Object.defineProperty(exports, "__esModule", {value: true});// src/index.ts
var _unplugin = require('unplugin');
var _pluginutils = require('@rollup/pluginutils');





var _common = require('@vue-macros/common');

// src/core/index.ts









function transformDefineProps(code, id) {
  if (!code.includes(_common.DEFINE_PROPS_DOLLAR))
    return;
  const { scriptSetup, getSetupAst } = _common.parseSFC.call(void 0, code, id);
  if (!scriptSetup)
    return;
  const offset = scriptSetup.loc.start.offset;
  const s = new (0, _common.MagicString)(code);
  const setupAst = getSetupAst();
  _common.walkAST.call(void 0, setupAst, {
    enter(node) {
      if (_common.isCallOf.call(void 0, node, _common.DEFINE_PROPS_DOLLAR)) {
        s.overwriteNode(
          node.callee,
          // add space for fixing mapping
          ` ${_common.DEFINE_PROPS}`,
          { offset }
        );
      }
    }
  });
  return _common.getTransformResult.call(void 0, s, id);
}

// src/index.ts
function resolveOption(options, framework) {
  const version = options.version || _common.detectVueVersion.call(void 0, );
  return {
    include: [_common.REGEX_VUE_SFC, _common.REGEX_SETUP_SFC].concat(
      version === 2 && framework === "webpack" ? _common.REGEX_VUE_SUB : []
    ),
    ...options,
    version
  };
}
var name = "unplugin-vue-define-props";
var src_default = _unplugin.createUnplugin.call(void 0, 
  (userOptions = {}, { framework }) => {
    const options = resolveOption(userOptions, framework);
    const filter = _pluginutils.createFilter.call(void 0, options.include, options.exclude);
    return {
      name,
      enforce: "pre",
      transformInclude(id) {
        return filter(id);
      },
      transform(code, id) {
        try {
          return transformDefineProps(code, id);
        } catch (err) {
          this.error(`${name} ${err}`);
        }
      }
    };
  }
);




exports.transformDefineProps = transformDefineProps; exports.src_default = src_default;
