// @vue-macros/define-props-refs@0.2.1/dist/chunk-ZSGEXC6R.js
"use strict";Object.defineProperty(exports, "__esModule", {value: true});// src/index.ts
var _unplugin = require('unplugin');
var _pluginutils = require('@rollup/pluginutils');





var _common = require('@vue-macros/common');

// src/core/index.ts










function transformDefinePropsRefs(code, id) {
  if (!code.includes(_common.DEFINE_PROPS_REFS))
    return;
  const { scriptSetup, getSetupAst } = _common.parseSFC.call(void 0, code, id);
  if (!scriptSetup)
    return;
  const offset = scriptSetup.loc.start.offset;
  const s = new (0, _common.MagicString)(code);
  const setupAst = getSetupAst();
  let changed = false;
  _common.walkAST.call(void 0, setupAst, {
    enter(node) {
      if (_common.isCallOf.call(void 0, node, _common.WITH_DEFAULTS) && node.arguments) {
        processDefinePropsRefs(node.arguments[0], node);
        this.skip();
      } else if (_common.isCallOf.call(void 0, node, _common.DEFINE_PROPS_REFS)) {
        processDefinePropsRefs(node);
      }
    }
  });
  if (changed) {
    s.prependLeft(offset, `
import { toRefs as _MACROS_toRefs } from 'vue'`);
  }
  return _common.getTransformResult.call(void 0, s, id);
  function processDefinePropsRefs(propsCall, defaultsCall) {
    let code2 = `${_common.DEFINE_PROPS}${s.slice(
      offset + propsCall.callee.end,
      offset + propsCall.end
    )}`;
    if (defaultsCall) {
      code2 = `${_common.WITH_DEFAULTS}(${code2}, ${s.sliceNode(
        defaultsCall.arguments[1],
        {
          offset
        }
      )})`;
    }
    s.prependLeft(offset, `
const __MACROS_props = ${code2}`);
    s.overwriteNode(
      defaultsCall || propsCall,
      "_MACROS_toRefs(__MACROS_props)",
      { offset }
    );
    changed = true;
  }
}

// src/index.ts
function resolveOption(options, framework) {
  const version = options.version || _common.detectVueVersion.call(void 0, );
  return {
    include: [_common.REGEX_VUE_SFC, _common.REGEX_SETUP_SFC].concat(
      version === 2 && framework === "webpack" ? _common.REGEX_VUE_SUB : []
    ),
    ...options,
    version
  };
}
var name = "unplugin-vue-define-props-refs";
var src_default = _unplugin.createUnplugin.call(void 0, 
  (userOptions = {}, { framework }) => {
    const options = resolveOption(userOptions, framework);
    const filter = _pluginutils.createFilter.call(void 0, options.include, options.exclude);
    return {
      name,
      enforce: "pre",
      transformInclude(id) {
        return filter(id);
      },
      transform(code, id) {
        try {
          return transformDefinePropsRefs(code, id);
        } catch (err) {
          this.error(`${name} ${err}`);
        }
      }
    };
  }
);




exports.transformDefinePropsRefs = transformDefinePropsRefs; exports.src_default = src_default;
